src/cdk/table/array-data-source.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DataSource} from './data-source';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+/**
+ * Array-based data source that can be used to connect to a CollectionViewer. Will trigger an
+ * update to the CollectionViewer whenever a new data array is set.
+ */
+export class ArrayDataSource<T> implements DataSource<T> {
```

Comment:
I'm wondering if we should move the data source into a separate cdk category from `table` since other components (mainly the tree) will use it as well.

@tinayuangao would you be able to use or extend this `ArrayDataSource` as-is for use in the tree?

I'm thinking we could move `DataSource`, `CollectionViewer` and probably the unrelated selection model into `cdk/collections`